argparse_dataclass.py

```diff
@@ -182,6 +182,15 @@ def _add_dataclass_options(self) -> None:
             if field.metadata.get("choices") is not None:
                 kwargs["choices"] = field.metadata["choices"]
 
+            if field.metadata.get("nargs") is not None:
+                kwargs["nargs"] = field.metadata["nargs"]
+                if field.metadata.get("type") is None:
+                    # When nargs is specified, field.type should be a list,
+                    # or something equivalent, like typing.List.
+                    # Using it would most likely result in an error, so the user
+                    # should specify the type of the elements within the list
+                    raise ValueError("A type must be specified along with nargs")
```

Comment:
Shouldn't we be able to extract the type `T` from `List[T]`?

Reply:
I thought about this too, it's possible using `typing.get_args`.
The reason I didn't do it is because it's not always possible. Say in the case where the type hint is a simple `list`, or even a custom type.

Reply:
If the annotation is just `list` I think it's fine to error out. I don't see any reason why custom types would fail here but I haven't tried it out yet.

Reply:
I guess I can try to extract it, and only raise an error when it can neither be extracted nor given explicitly.
Also, assuming a user both uses a parameterized type hint (i.e. `List[T]`) and provides a `type` through `field.metadata`, I believe the latter should take priority. This would allow for example:
```python
names: List[str] = field(metadata=dict(nargs='*', type=str.lower))  # convert all names in the list to lowercase
```
Let me know what you think

Reply:
Yeah I agree that overrides can take precedence. I would however recommend that if you want to do something like `str.lower` you do that in a `__post_init__` method instead of as shown (but that's just my personal preference).

Reply:
Of course, this was just for the sake of an example. A more sophisticated one could be:
```python
files: List[TextIO] = field(metadata=dict(nargs='*', type=argparse.FileType('r')))
```
i.e. it accepts a list of files and opens them for reading.

Anyway, I just pushed a commit that tries to extract the type before raising an error.